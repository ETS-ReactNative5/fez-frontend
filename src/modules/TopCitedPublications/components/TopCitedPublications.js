import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {locale} from 'locale';
import {Tabs, Tab} from 'material-ui/Tabs';
import {StandardCard} from 'modules/SharedComponents/Toolbox/StandardCard';
import {InlineLoader} from 'modules/SharedComponents/Toolbox/Loaders';
import {PublicationsList} from 'modules/SharedComponents/PublicationsList';
import {HelpIcon} from 'modules/SharedComponents/Toolbox/HelpDrawer';
import Alert from 'modules/SharedComponents/Toolbox/Alert/components/Alert';

export default class TopCitedPublications extends PureComponent {
    static contextTypes = {
        isMobile: PropTypes.bool
    };

    static propTypes = {
        topCitedPublicationsList: PropTypes.array,
        loadingTopCitedPublications: PropTypes.bool,
        actions: PropTypes.object.isRequired,
        showSourceCountIcon: PropTypes.bool
    };

    static defaultProps = {
        topCitedPublicationsList: [],
        loadingTopCitedPublications: false
    };

    componentDidMount() {
        if (!this.props.loadingTopCitedPublications) {
            this.props.actions.searchTopCitedPublications();
        }
    }

    render() {
        const txt = locale.components.topCitedPublications;

        if (this.props.loadingTopCitedPublications) {
            return (
                <div className="isLoading is-centered">
                    <InlineLoader message={txt.loading}/>
                </div>
            );
        }

        return (
            <div className={'topCitedPublications'}>
                {!this.props.loadingTopCitedPublications && this.props.topCitedPublicationsList.length > 0 ?
                    <StandardCard>
                        <Tabs className="publicationTabs" inkBarStyle={{height: '4px', marginTop: '-4px'}}>
                            {
                                this.props.topCitedPublicationsList.sort((source1, source2) => (
                                    txt[source1.key].order - txt[source2.key].order
                                )).map(({key, values}, metricIndex) => (
                                    <Tab key={key}
                                        label={this.context.isMobile ? txt[key].mobileTitle : txt[key].title}
                                        value={`${key}TopCitedPublications`}
                                        className="publicationTab"
                                    >
                                        <div className="publicationTabContent">
                                            <div>
                                                <div className="is-pulled-right">
                                                    <HelpIcon {...locale.components.trendingPublicationHelp}/>
                                                </div>
                                                <div key={'metrics_' + metricIndex} className="trendingPubsSection">
                                                    <h2 className="trendingPubsSource">
                                                        <div className={`fez-icon ${key} xxlarge`}/>
                                                        {txt[key].heading}
                                                    </h2>
                                                    <div className="is-hidden-mobile subTitle">{txt[key].subHeading}</div>
                                                    <PublicationsList
                                                        publicationsList={values}
                                                        showMetrics
                                                        showSourceCountIcon={key === 'altmetric'}
                                                        hideCountDiff={key === 'altmetric'}
                                                        hideCitationContent
                                                        missingText={txt.missing}
                                                    />
                                                </div>
                                            </div>
                                        </div>
                                    </Tab>
                                ))
                            }
                        </Tabs>
                    </StandardCard>
                    :
                    <Alert {...txt.notAvailableAlert} />
                }
            </div>
        );
    }
}


