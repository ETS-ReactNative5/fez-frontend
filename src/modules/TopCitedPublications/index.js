export {default as TopCitedPublications} from './containers/TopCitedPublications';
