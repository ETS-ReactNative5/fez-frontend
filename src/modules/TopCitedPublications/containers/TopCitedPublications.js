import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import TopCitedPublications from '../components/TopCitedPublications';
import * as actions from 'actions';
import {withRouter} from 'react-router-dom';

const mapStateToProps = (state) => {
    return {
        ...state.get('topCitedPublicationsReducer')
    };
};

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(actions, dispatch)
    };
}

let TopCitedPublicationsContainer = connect(mapStateToProps, mapDispatchToProps)(TopCitedPublications);
TopCitedPublicationsContainer = withRouter(TopCitedPublicationsContainer);

export default TopCitedPublicationsContainer;
