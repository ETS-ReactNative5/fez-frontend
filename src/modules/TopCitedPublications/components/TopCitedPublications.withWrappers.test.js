import TopCitedPublications from './TopCitedPublications';

function setup(testProps, isShallow = true){
    const props = {
        actions: {
            searchTopCitedPublications: jest.fn()
        },
        ...testProps
    };
    return getElement(TopCitedPublications, props, isShallow);
}

describe('Component TopCitedPublications with wrappers', () => {
    it('should render properly', () => {
        const wrapper = setup({}, false);
        expect(toJson(wrapper.instance())).toMatchSnapshot();
    });
});