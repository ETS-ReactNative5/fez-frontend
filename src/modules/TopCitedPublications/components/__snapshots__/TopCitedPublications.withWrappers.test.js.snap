// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TopCitedPublications with wrappers should render properly 1`] = `undefined`;
